theforeman.org/scripts/debian/configure_pbuilder.sh

```diff
@@ -4,29 +4,30 @@ set -xe
 # This script sets up the PBuilder hooks to pull in our deb repositories while the package is building
 echo "--Apt Hook Setup"
 
-for pbuilderd in /etc/pbuilder/${os}*; do
-  # Cleanup any old hooks
-  sudo rm -f ${pbuilderd}/hooks/F60addforemanrepo
+pdbuilderd=/etc/pbuilder/${os}64
 
+# Cleanup any old hooks
+sudo rm -f ${pbuilderd}/hooks/F60addforemanrepo
+
+if [[ $repoowner != theforeman ]]; then
+  # Add a stagingdeb repo to the sources too, if it exists
   if wget -O/dev/null -q http://stagingdeb.theforeman.org/dists/${os}/${repoowner}-${version} ; then
     echo "echo deb http://stagingdeb.theforeman.org/ ${os} ${repoowner}-${version} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
   fi
   if wget -O/dev/null -q http://stagingdeb.theforeman.org/dists/plugins/${repoowner} ; then
     echo "echo deb http://stagingdeb.theforeman.org/ plugins ${repoowner} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
   fi
+fi
 
-  # Use tee to get around sudo/redirection problems, with /dev/null to drop the stdout
-  echo "echo deb http://deb.theforeman.org/ ${os} ${version} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
-  echo "echo deb http://deb.theforeman.org/ plugins ${version} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
+# Use tee to get around sudo/redirection problems, with /dev/null to drop the stdout
+echo "echo deb http://deb.theforeman.org/ ${os} ${version} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
+echo "echo deb http://deb.theforeman.org/ plugins ${version} >> /etc/apt/sources.list" | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo > /dev/null
 
-  # Permit per-OS/project setups to add their own repos
-  if [ -e debian/${os}/${project}/hooks ]; then
-    sed "s/%OS%/${os}/g" debian/${os}/${project}/hooks | sudo tee -a ${pbuilderd}/hooks/F60addforemanrepo
-  fi
+# Read Foreman package installation logs on failure
+echo "cat /var/log/foreman-install.log" || true | sudo tee ${pbuilderd}/hooks/C10foremanlog
+
+# Make executable
+sudo chmod 0775 ${pbuilderd}/hooks/*
```

Comment:
If you're going this route, then I'd move it below `F99printrepos` so you can skip one chmod.

However, I wonder if there are any example hooks in there already we don't want to activate. Then again, it already was done in plugin so it's probably safe.

Reply:
I don't know enough to know what use case this covered but it was in a few of the scripts so I kept it.